### PR TITLE
Lazily import openfermion

### PR DESCRIPTION
**Pull request by trbromley**
Prevents `openfermion`, `openfermionpsi4` and `openfermionpyscf` from being automatically imported with PennyLane.

For `openfermion`, we create a helper module with `getattr()` defined such that calls to e.g. `openfermion.ops.FermionOperator` are routed through the helper module.

For `openfermionpsi4` and `openfermionpyscf`, since they are only used in one spot we just keep things simple and include the imports at that point.

Timings:
`time python -c "import pennylane"` using this branch:
```
real    0m0.940s
user    0m1.277s
sys     0m1.673s
```
Using `master`:
```
real    0m2.154s
user    0m2.391s
sys     0m1.818s
```

Note that this supersedes https://github.com/PennyLaneAI/pennylane/pull/1284.

### CODE REVIEW

**Comment by trbromley** [comment]
Not sure if we need to update the changelog for this one :thinking:

File: qchem/pennylane_qchem/qchem/obs.py [hunk 1]
```diff
@@ -314,9 +316,13 @@ def observable(fermion_ops, init_term=0, mapping="jordan_wigner", wires=None):
 
     # Map the fermionic operator to a qubit operator
     if mapping.strip().lower() == "bravyi_kitaev":
-        return structure.convert_observable(bravyi_kitaev(mb_obs), wires=wires)
+        return structure.convert_observable(
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem/obs.py:319]: Oh.. is the qchem folder not being checked by black?

**Review by josh146 — APPROVED** [review]
This is a nice solution @trbromley! Probably cleaner than lazy importing QChem as a whole.

File: qchem/pennylane_qchem/qchem/obs.py (hunk 1 above)
Comment (agran2018) [reply qchem/pennylane_qchem/qchem/obs.py:319]: Yes, this is weird. Perhaps using an outdated version of black?

Comment (trbromley) [reply qchem/pennylane_qchem/qchem/obs.py:315]: I don't think qchem and the qchem tests are being automatically checked with `black` - do we want to turn that on (I vote yes)?

**Review by agran2018 — APPROVED** [review]
Thanks @trbromley!, looks good to me. I have noted that there is a Codecov warning for Line 647. But it seems that it is a format issue.

File: qchem/pennylane_qchem/qchem/obs.py (hunk 1 above)
Comment (josh146) [reply qchem/pennylane_qchem/qchem/obs.py:315]: I vote yes as well, I think it was an oversight that it wasn't being checked!

**Pull request merged**